### ISSUE TITLE
Run project test suite on CI system

### ISSUE THREAD

**Issue by jryans** [stephenrkell/liballocs#26]
As this is a complex project with various interlocking components and sub-repos, I think it would be quite helpful to have tests running in some kind of CI system for each PR.

Since we already have the `buildtest` Docker files around, we would need to experiment with running that on some CI system and hooking it up to run for each PR. It's possible we may run up against resource limits on free CI tools if it takes a while to build and run.

**Comment by stephenrkell** [comment]
Yep... it's been on my list for a while. I did have a liballocs config on a Jenkins box in Cambridge but that was during a quiet time. I'm told we have a CI system here at Kent that I can use. I will investigate.

(A related problem is test cases. Since liballocs for a long time had only one client, libcrunch, I historically relied on libcrunch's tests to test liballocs. That's no longer a good idea, but getting similar coverage more freestandingly will take some effort.)

**Comment by jryans** [comment]
I was considering sending some PRs to use [CircleCI](https://circleci.com/), which is generally well-reviewed among OSS projects. It's free for open source projects.

It seems like an relatively easy way to get some CI in place (which would massively boost my own confidence with the code as someone new to these projects), and you can always migrate to an internal system later if desired.

Would you be open to that?

**Comment by stephenrkell** [comment]
Sorry for the delay. Short answer: yes! That would be great. I don't want it to be The Permanent Solution but that's no reason not to do it for now.

Currently there are two or three liballocs test cases that are known (er, to me) to be failing. Will be interesting to see if we agree on which ones. :-)
